spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala

```diff
@@ -0,0 +1,654 @@
+            val attrValue = AttrValue.newBuilder
+            attrValue.setDataType(DataType.TENSOR)
+            attrValue.setTensorValue(tensor)
+            tensors(i) = TensorConverter.
+              getAttributeValue(attrValue.build).asInstanceOf[Tensor[T]]
+            i += 1
+          })
+          tensors
+        case DataType.VARIABLE_FORMAT =>
+          val formats = new Array[VariableFormat](size)
+          val formatList = valueArray.getVariableFormatList.asScala
+          var i = 0
+          formatList.foreach(format => {
+            val attrValue = AttrValue.newBuilder
+            attrValue.setDataType(DataType.VARIABLE_FORMAT)
+            attrValue.setVariableFormatValue(format)
+            formats(i) = VariableFormatConverter.
+              getAttributeValue(attrValue.build).asInstanceOf[VariableFormat]
+          })
+          formats
+        case DataType.INITMETHOD =>
+          val methods = new Array[InitializationMethod](size)
+          val methodList = valueArray.getInitMethodList.asScala
+          var i = 0
+          methodList.foreach(method => {
+            val attrValue = AttrValue.newBuilder
+            attrValue.setDataType(DataType.INITMETHOD)
+            attrValue.setInitMethodValue(method)
+            methods(i) = InitMethodConverter.getAttributeValue(attrValue.build)
+            .asInstanceOf[InitializationMethod]
+            i += 1
+          })
+          methods
+        case DataType.MODULE =>
+          val modules = new Array[AbstractModule[Activity, Activity, T]](size)
+          val moduleList = valueArray.getBigDLModuleList.asScala
+          var i = 0
+          moduleList.foreach(module => {
+            val attrValue = AttrValue.newBuilder
+            attrValue.setDataType(DataType.MODULE)
+            attrValue.setBigDLModuleValue(module)
+            modules(i) = ModuleConverter.
+              getAttributeValue(attrValue.build).asInstanceOf[AbstractModule[Activity, Activity, T]]
+            i += 1
+          })
+          modules
+        case DataType.NAME_ATTR_LIST =>
+          val nameArray = new Array[Map[String, Map[String, Any]]](size)
+          val nameAttriLists = valueArray.getNameAttrListList.asScala
+          val i = 0
+          nameAttriLists.foreach(nameList => {
+            val attrValue = AttrValue.newBuilder
+            attrValue.setDataType(DataType.NAME_ATTR_LIST)
+            attrValue.setNameAttrListValue(nameList)
+            nameArray(i) = NameListConverter.getAttributeValue(attrValue.build)
+              .asInstanceOf[Map[String, Map[String, Any]]]
+          })
+          nameArray
+      }
+      arr
+    }
+
+    override def setAttributeValue[T: ClassTag](attributeBuilder: AttrValue.Builder,
+      value: Any, valueType: universe.Type = null)(implicit ev: TensorNumeric[T]): Unit = {
+      attributeBuilder.setDataType(DataType.ARRAY_VALUE)
+      val arrayBuilder = ArrayValue.newBuilder
+      if (value.isInstanceOf[Array[Int]]) {
```

Comment:
Yes it works , Array's behaviors different from other containers , so it doesn't work if it's a list.